src/crews.js

```diff
@@ -1,49 +1,127 @@
 const { execQuery } = require('./db');
 const moment = require('moment');
 
-const CREW_GET = `
+const CREW_GET_CC = `
     SELECT
-        CONCAT(SUBSTRING(c.first_name, 1, 1), '. ', c.last_name) as cc,
-        CONCAT(SUBSTRING(d.first_name, 1, 1), '. ', d.last_name) as driver,
-        CONCAT(SUBSTRING(r1.first_name, 1, 1), '. ', r1.last_name) as rider1,
-        CONCAT(SUBSTRING(r2.first_name, 1, 1), '. ', r2.last_name) as rider2,
-        CONCAT(SUBSTRING(ds.first_name, 1, 1), '. ', ds.last_name) as dutysup,
-        c.radionum as ccrn,
-        d.radionum as driverrn,
-        r1.radionum as rider1rn,
-        r2.radionum as rider2rn,
-        ds.radionum as dutysuprn
+        c.cc as id,
+        CONCAT(SUBSTRING(c.first_name, 1, 1), '. ', c.last_name) as name,
+        c.radionum as rn
     FROM
-        (SELECT m.first_name, m.last_name, m.radionum, c.date FROM crews c, members m
-            WHERE m.id = c.cc AND c.date = ?) as c,
-        (SELECT m.first_name, m.last_name, m.radionum, c.date FROM crews c, members m
-            WHERE m.id = c.driver AND c.date = ?) as d,
-        (SELECT m.first_name, m.last_name, m.radionum, c.date FROM crews c, members m
-            WHERE m.id = c.attendant AND c.date = ?) as r1,
-        (SELECT m.first_name, m.last_name, m.radionum, c.date FROM crews c, members m
-            WHERE m.id = c.observer AND c.date = ?) as r2,
-        (SELECT m.first_name, m.last_name, m.radionum, c.date FROM crews c, members m
+        (SELECT c.cc, m.first_name, m.last_name, m.radionum, c.date FROM crews c, members m
+            WHERE m.id = c.cc AND c.date = ?) as c
+    WHERE
+        c.date IS NOT NULL;
+`;
+
+
+const CREW_GET_D = `
+    SELECT
+        d.driver as id,
+        CONCAT(SUBSTRING(d.first_name, 1, 1), '. ', d.last_name) as name,
+        d.radionum as rn
+    FROM
+        (SELECT c.driver, m.first_name, m.last_name, m.radionum, c.date FROM crews c, members m
+            WHERE m.id = c.driver AND c.date = ?) as d
+    WHERE
+        d.date IS NOT NULL;
```

Comment:
If you're going to unwind these into their own DB queries, then you can remove the subquery and get slightly better performance just using a regular query, so something like:
```
    SELECT
        c.driver AS id
        CONCAT(SUBSTRING(m.first_name, 1, 1), '. ', m.last_name) as name,
        m.radionum AS rn
    FROM
        crews AS c
    INNER JOIN members AS m ON m.id = c.driver
    WHERE c.date = ?
```

I'd also wrap this in a function so you could do:
```
function getQuery(role) {
    return `SELECT
        c.${role} AS id
        CONCAT(SUBSTRING(m.first_name, 1, 1), '. ', m.last_name) as name,
        m.radionum AS rn
    FROM
        crews AS c
    INNER JOIN members AS m ON m.id = c.${role}
    WHERE c.date = ?`;
}
```

and then call the function with `getQuery('driver')` or `getQuery('dutysup')` or etc so that way we've only got this thing written once in case want to make changes to it in the future. You could then wrap calling this in conjunction with executing the query to further reduce repetition.

Reply:
Thanks so much for the thoughts - I'll definitely get on implementing that.